aleph/tests/test_bookmarks_api.py

```diff
@@ -0,0 +1,113 @@
+from aleph.model import Bookmark
+from aleph.core import db
+
+
+class BookmarksApiTestCase(TestCase):
+    def setUp(self):
+        super(BookmarksApiTestCase, self).setUp()
+
+        self.role, self.headers = self.login()
+        self.collection = self.create_collection(self.role, label="Politicians")
+
+        data = {"schema": "Person", "properties": {"name": "Angela Merkel"}}
+        self.entity = self.create_entity(data=data, collection=self.collection)
+        index_entity(self.entity)
+
+    def test_bookmarks_index_auth(self):
+        res = self.client.get("/api/2/bookmarks")
+        assert res.status_code == 403, res
+
+        res = self.client.get("/api/2/bookmarks", headers=self.headers)
+        assert res.status_code == 200, res
+        assert res.json["total"] == 0, res.json
+
+    def test_bookmarks_index_results(self):
+        bookmark = Bookmark(entity_id=self.entity.id, role_id=self.role.id)
+        db.session.add(bookmark)
+        db.session.commit()
+
+        res = self.client.get("/api/2/bookmarks", headers=self.headers)
+        assert res.status_code == 200, res
+        assert res.json["total"] == 1, res.json
+        entity = res.json["results"][0]["entity"]
+        assert entity["properties"]["name"][0] == "Angela Merkel", res.json
+
+    def test_bookmarks_index_read_permission(self):
+        """Only returns bookmarks if user has access to referenced entity"""
+        other_role = self.create_user(foreign_id="other")
+        secret_collection = self.create_collection(other_role, label="Top Secret")
+
+        data = {"schema": "Person", "properties": {"name": ["Mister X"]}}
+        secret_entity = self.create_entity(data, secret_collection)
+        index_entity(secret_entity)
+
+        bookmark = Bookmark(entity_id=secret_entity.id, role_id=self.role.id)
+        db.session.add(bookmark)
+        db.session.commit()
+
+        res = self.client.get("/api/2/bookmarks", headers=self.headers)
+        assert res.status_code == 200, res
+        assert res.json["total"] == 0, res.json
+        assert len(res.json["results"]) == 0, res.json
+
+    def test_bookmarks_create(self):
+        count = Bookmark.query.count()
+        assert count == 0, count
+
+        body = [{"entity_id": self.entity.id}]
+        res = self.client.post(
+            "/api/2/bookmarks",
+            headers=self.headers,
+            data=json.dumps(body),
+            content_type=JSON,
+        )
+        assert res.status_code == 201, res
+        assert len(res.json) == 1, res.json
+        entity = res.json[0]["entity"]
+        assert entity["properties"]["name"][0] == "Angela Merkel", res.json
+
+        count = Bookmark.query.count()
+        assert count == 1, count
+        bookmark = Bookmark.query.first()
+        assert bookmark.entity_id == self.entity.id, bookmark.entity_id
+        assert bookmark.role_id == self.role.id, bookmark.role_id
+
+    def test_bookmarks_create_validate_entity_id(self):
```

Comment:
This test is currently failing. Users can currently create bookmarks for random IDs without validations.